stdlib/public/core/UnsafeBufferPointer.swift.gyb

```diff
@@ -232,6 +232,15 @@ public struct Unsafe${Mutable}BufferPointer<Element>
 %  end
   }
 
+  /// Creates an empty instance.
+  public init() {
+    // This is known to create a suitably-aligned pointer on all
+    // currently-supported platforms.
```

Comment:
I think that constrains the runtime's assumptions about null's bitpattern. I'll have to think about a counter proposal.